### PR TITLE
[Feature] Support lsq

### PR DESCRIPTION
**Pull request by HIT-cwh**
# Modifications

1. Support deploy_cfg is None
2. Put `replace_fakequant` before 'load_state_dict` (for lsq)
3. Add `_load_from_state_dict` interface to lsq fakequant to fix the bug when loading ckpt in mmdeploy.
4. Add pytest
5. change github ci: ubuntu 18.04 to ubuntu 20.04

### CODE REVIEW

File: mmrazor/models/algorithms/quantization/mm_architecture.py [hunk 1]
```diff
@@ -354,10 +363,13 @@ def get_deploy_model(self):
         fp32_model = self.architecture
         self.quantizer.convert_batchnorm2d(fp32_model)
         observed_model = self.quantizer.prepare(fp32_model)
-        observed_model.load_state_dict(quantized_state_dict)
+        observed_model.load_state_dict(quantized_state_dict, strict=False)
```
Comment (humu789) [inline mmrazor/models/algorithms/quantization/mm_architecture.py:366]: why `strict` defaults to `false`

**Review by humu789 — COMMENTED** [review]
1. need to verify common qat pipeline and accuracy
2. need to pass CI test

File: mmrazor/models/algorithms/quantization/mm_architecture.py (hunk 1 above)
Comment (HIT-cwh) [reply mmrazor/models/algorithms/quantization/mm_architecture.py:366]: Fixed!
